tools/test/windows/tw.cc

```diff
@@ -1719,6 +1772,101 @@ Path Path::Dirname() const {
   return result;
 }
 
+IFStream* IFStreamImpl::Create(bazel::windows::AutoHandle* handle,
+                               DWORD max_page_size) {
+  std::unique_ptr<uint8_t[]> data(new uint8_t[max_page_size * 2]);
+  DWORD read;
+  if (!ReadFile(*handle, data.get(), max_page_size * 2, &read, NULL)) {
+    DWORD err = GetLastError();
+    if (err == ERROR_BROKEN_PIPE) {
+      read = 0;
+    } else {
+      LogErrorWithValue(__LINE__, "Failed to read from file", err);
+      return nullptr;
+    }
+  }
+  return new IFStreamImpl(handle, std::move(data), read, max_page_size);
+}
+
+bool IFStreamImpl::Get(uint8_t* result) const {
+  if (read_pos_ < page_end_) {
```

Comment:
Did you consider just calling PeekN(0, result) here?

Reply:
Now I did. It's an interesting idea but doesn't bring any benefit for my use-case.

Supporting this requires a slight adjustment to PeekN, because it now reads from `read_pos_+1`. With the adjustment it could read from the current cursor position, at which point I could drop Get().

However the way I'll use the IFStream is to Get() the current byte, then decide how many more to Peek at (if any), then decide how many more to Advance. (Use-case: UTF-8 escaping.) Changing PeekN to support 0-byte lookahead (i.e. replace Get() with Peek0) would mean that PeekN would read the current byte again, and the N subsequent ones. But the current one is already read, so there's no reason to read it again.